### ISSUE TITLE
Coffeescript outputs different javascript in different servers

### ISSUE THREAD

**Issue by nisanth074** [jashkenas/coffeescript#4182]
Hi,

I ran into an issue where Coffeescript (I'm using version `1.10.10`) outputs different javascript for the same input in different machines. Here are the steps to replicate the issue

Install the `coffee-script` gem

```
gem install coffee-script -v "2.2.0"
```

Open a ruby repl

```
~/code% irb
1.9.3-p484 :001 >
```

and copy paste the following code snippet in the repl

```ruby
require 'coffee-script'

coffee = <<COFFEESCRIPT
CoffeeScript.VERSION
COFFEESCRIPT
puts CoffeeScript::Source.context.eval(coffee)

coffee = <<COFFEESCRIPT
class Dog extends Animal
COFFEESCRIPT
js = CoffeeScript.compile(coffee)
puts js
```

On one machine, the javascript generated is

```javascript
(function() {
  var Dog,
    extend = function(child, parent) { for (var key in parent) { if (hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; },
    hasProp = {}.hasOwnProperty;

  Dog = (function(superClass) {
    extend(Dog, superClass);

    function Dog() {
      return Dog.__super__.constructor.apply(this, arguments);
    }

    return Dog;

  })(Animal);

}).call(this);
```

On an other machine, the javascript generated is

```javascript
(function() {
  var Dog,
    hasProp = {}.hasOwnProperty,
    extend = function(child, parent) { for (var key in parent) { if (hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; };

  Dog = (function(superClass) {
    extend(Dog, superClass);

    function Dog() {
      return Dog.__super__.constructor.apply(this, arguments);
    }

    return Dog;

  })(Animal);

}).call(this);
```

The rails framework uses a fingerprinting strategy to cache assets. The fingerprint of an asset (like a javascript asset) is calculated from the contents of the file. Since the javascript generated by coffeescript is different in different servers, the fingerprints vary across servers for the same deploy. These inconsistent fingerprints are causing numerous issues.

Why is this happening? If it helps to investigate, we were using Coffeescript `1.3.3` earlier which generated javascript consistently. Thank you.

**Comment by JimPanic** [comment]
What is the difference between those machines? Have you tried compiling the same code on both machines without Ruby, but only with coffee itself? Are there differences as well when you compile with the CLI instead of inside irb?

**Comment by nisanth074** [comment]
@JimPanic Thanks for the quick reply. The issue happens even when compiling directly with coffee 1.10.10 (but not with 1.3.3, the coffee version we used earlier) from CLI. I eventually found that the affected servers were using node 0.11.2-pre, while the other servers used node 0.6.17. I upgraded all servers to use a fairly recent and stable version of node and the issue was fixed.

I'm not sure why we faced this issue with coffee 1.10.10, but not with coffee 1.3.3
